repos/system_upgrade/common/models/rhuiinfo.py

```diff
@@ -1,12 +1,55 @@
-from leapp.models import fields, Model
+from leapp.models import CopyFile, fields, Model
 from leapp.topics import SystemInfoTopic
 
 
+class TargetRHUIPreInstallTasks(Model):
+    """Tasks required to be executed before target RHUI clients are installed"""
+    topic = SystemInfoTopic
+
+    files_to_remove = fields.List(fields.String(), default=[])
+    """Files to remove from the source system in order to setup target RHUI access"""
+
+    files_to_copy_in = fields.List(fields.Model(CopyFile), default=[])
+    """Files to copy into the scratch (overlayfs) container in order to setup target RHUI access"""
+
+
+class TargetRHUIPostInstallTasks(Model):
+    """Tasks required to be executed after target RHUI clients are installed to facilitate access to target content."""
+    topic = SystemInfoTopic
+
+    files_to_copy = fields.List(fields.Model(CopyFile), default=[])
+    """Source and destination are paths inside the container"""
+
+
+class TargetRHUISetupInfo(Model):
+    topic = SystemInfoTopic
+
+    enable_only_repoids_in_copied_files = fields.Boolean(default=True)
+    """If True (default) only the repoids from copied files will be enabled during client installation"""
+
+    preinstall_tasks = fields.Model(TargetRHUIPreInstallTasks)
+    """Tasks that must be performed before attempting to install the target client(s)"""
+
+    postinstall_tasks = fields.Model(TargetRHUIPostInstallTasks)
+    """Tasks that must be performed after the target client is installed (before any other content is accessed)"""
+
+    are_clients_sufficient = fields.Boolean(default=True)
+    """True if clients provide everything necessary to access target RHUI content"""
+
+
 class RHUIInfo(Model):
     """
     Facts about public cloud provider and RHUI infrastructure
     """
     topic = SystemInfoTopic
 
     provider = fields.String()
-    """ Provider name """
+    """Provider name"""
```

Comment:
OK, I agree with you that this might be the time to correct the history and properly distinguish between providers and system variants - have separate fields for this.

Reply:
Done.